src/go/rdctl/cmd/snapshot.go

```diff
@@ -16,6 +15,14 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type errorPayloadType struct {
+	Error string `json:"error,omitempty"`
+}
+
+var outputJsonFormat bool
+var snapshotErrors []error
```

Comment:
I'd prefer to avoid global variables if possible. What if we changed the function signature of `exitWithJsonOrErrorCondition()` to:
```
func exitWithJsonOrErrorCondition(errs ...error) error { ... }
```
and then returned a `[]error` from `wrapSnapshotOperation()`, one for each of `functionErr`, `startVMErr` and `waitForStartedErr`? Which could be passed to `exitWithJsonOrErrorCondition()`?

Reply:
The `outputJsonFormat` bool is global because most of the subcommands support a `--json` argument that changes it.

As for the error array, does it even make sense to continue with the update-backend-state and wait-for-vm-state if the wrapped function returns an error condition? Because if we fail on one of them, then we don't need to carry an array of error objects -- it's the go way to return on failure as soon as possible.

Reply:
> The outputJsonFormat bool is global because most of the subcommands support a --json argument that changes it.

👍 

> As for the error array, does it even make sense to continue with the update-backend-state and wait-for-vm-state if the wrapped function returns an error condition? Because if we fail on one of them, then we don't need to carry an array of error objects -- it's the go way to return on failure as soon as possible.

The idea is to try to leave the main process in a usable state even if there is an error in the snapshot operation. That means we should try to ensure the backend is restarted in the event of an error in the snapshot operation.

Of course, there's more to the story. This block of code will probably be changed as part of:
- #5643
- #5684
- #5691

And maybe others. I'd say we should just make the changes relevant to this PR, and keep the rest as unchanged as possible in order to ease rebases.